File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        i=0
        j=len(nums)-1
        k=len(nums)-1
        res=[0]*len(nums)
        while(i<=j):
            if(abs(nums[i])<abs(nums[j])):
                res[k]=nums[j]*nums[j]
                k-=1
                j-=1
            else:
                res[k]=nums[i]*nums[i]
                k-=1
                i+=1
        return res

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* sortedSquares(int* nums, int numsSize, int* returnSize) {
        int i=0;
        int j=numsSize-1;
        int k=numsSize-1;
        int *res=(int*)malloc(numsSize*sizeof(int));
        while(i<=j)
        {
            if(abs(nums[i])<abs(nums[j]))
            {
                res[k]=nums[j]*nums[j];
                k--;
                j--;
            }
            else
            {
                res[k]=nums[i]*nums[i];
                k--;
                i++;
            }
        }
        *returnSize=numsSize;
        return res;
}